docs/examples/201/vm-new-or-existing-conditions/main.bicep

```diff
@@ -0,0 +1,272 @@
+param location string {
+  metadata: {
+    description: 'Location to for the resources.'
+  }
+  default: resourceGroup().location
+}
+param vmName string {
+  metadata: {
+    description: 'Name for the Virtual Machine.'
+  }
+  default: 'linux-vm'
+}
+param adminUsername string {
+  metadata: {
+    description: 'User name for the Virtual Machine.'
+  }
+}
+param authenticationType string {
+  allowed: [
+    'password'
+    'sshPublicKey'
+  ]
+  metadata: {
+    description: 'Type of authentication to use on the Virtual Machine.'
+  }
+  default: 'sshPublicKey'
+}
+param adminPasswordOrKey string {
+  metadata: {
+    description: 'Password or ssh key for the Virtual Machine.'
+  }
+  secure: true
+}
+param vmSize string {
+  metadata: {
+    description: 'Size for the Virtual Machine.'
+  }
+  default: 'Standard_A2_v2'
+}
+param storageNewOrExisting string {
+  metadata: {
+    description: 'Determines whether or not a new storage account should be provisioned.'
+  }
+  default: 'new'
+}
+param storageAccountName string {
+  metadata: {
+    description: 'Name of the storage account'
+  }
+  default: 'storage${uniqueString(resourceGroup().id)}'
+}
+param storageAccountType string {
+  metadata: {
+    description: 'Storage account type'
+  }
+  default: 'Standard_LRS'
+}
+param storageAccountResourceGroupName string {
+  metadata: {
+    description: 'Name of the resource group for the existing storage account'
+  }
+  default: resourceGroup().name
+}
+param virtualNetworkNewOrExisting string {
+  metadata: {
```

Comment:
I think we should use `allowed` on this field to turn it into an enum, or turn it into a bool (same for other `..NewOrExisting` params:

```bicep
param virtualNetworkNewOrExisting string {
  metadata: {
    description: 'Determines whether or not a new virtual network should be provisioned.'
  }
  allowed: [
    'new'
    'existing'
  ]
  default: 'new'
}
```

Reply:
Turned those into bool parameters.